### PR TITLE
more info (optional) taking while getting an API key

### PR DESCRIPTION
**Pull request by rex-god**

### CODE REVIEW

File: src/dal/user.schema.ts [hunk 1]
```diff
@@ -11,6 +11,18 @@ export class User {
...
   api_key: string;
 
+  @Prop({ required: false })
+  name: string;
+
+  @Prop({ required: false })
+  team_size: string;
+
+  @Prop({ required: false })
+  organization: string;
```
Comment (vgvishesh) [inline src/dal/user.schema.ts:21]: rename this to `project_name`

File: src/dal/user.schema.ts [hunk 2]
```diff
@@ -11,6 +11,18 @@ export class User {
...
+
+  @Prop({ required: false })
+  building: string;
```
Comment (vgvishesh) [inline src/dal/user.schema.ts:24]: rename this to `project_info`

File: src/dto/get-api-key.dto.ts [hunk 3]
```diff
@@ -1,12 +1,46 @@
...
   readonly email: string;
+
+  @ApiProperty({
+    title: 'organization',
+    description: 'Your organization name',
+    example: 'ABC Project',
+  })
+  @IsOptional()
+  @Length(3, 30)
+  readonly organization?: string;
```
Comment (vgvishesh) [inline src/dto/get-api-key.dto.ts:29]: rename to project_name

File: src/dto/get-api-key.dto.ts [hunk 4]
```diff
@@ -1,12 +1,46 @@
...
+  @IsOptional()
+  readonly team_size?: string;
+
+  @ApiProperty({
+    title: 'building',
+    description: 'What you are building',
+    example: 'We are building...',
+  })
+  @IsOptional()
+  readonly building?: string;
```
Comment (vgvishesh) [inline src/dto/get-api-key.dto.ts:45]: rename to project_info

File: src/dto/get-api-key.dto.ts [hunk 5]
```diff
@@ -1,12 +1,46 @@
...
+  @Length(3, 30)
+  readonly organization?: string;
+
+  @ApiProperty({
+    title: 'team_size',
+    description: 'Your team size',
+    example: '20-50',
+  })
+  @IsOptional()
+  readonly team_size?: string;
```
Comment (vgvishesh) [inline src/dto/get-api-key.dto.ts:37]: what are the other options for `team_size`?
 Ideally, this should be a number,

Comment (rex-god) [reply src/dto/get-api-key.dto.ts:37]: cause this gonna like 1-10, 10-50, that's why it kept as a string.

File: src/dto/get-api-key.dto.ts (hunk 4 above)
Comment (rex-god) [reply src/dto/get-api-key.dto.ts:45]: ok

File: src/dto/get-api-key.dto.ts (hunk 3 above)
Comment (rex-god) [reply src/dto/get-api-key.dto.ts:29]: ok

File: src/dal/user.schema.ts (hunk 2 above)
Comment (rex-god) [reply src/dal/user.schema.ts:24]: ok

File: src/dal/user.schema.ts (hunk 1 above)
Comment (rex-god) [reply src/dal/user.schema.ts:21]: ok

File: src/dto/get-api-key.dto.ts (hunk 5 above)
Comment (vgvishesh) [reply src/dto/get-api-key.dto.ts:37]: how many such options are there and what are their values?

File: src/dto/get-api-key.dto.ts [hunk 6]
```diff
@@ -1,12 +1,46 @@
...
+  readonly name?: string;
+
   @ApiProperty({
     title: 'email',
     description: 'Your Email ID',
     example: 'john.doe@company.in',
   })
   @IsEmail()
   readonly email: string;
+
```
Comment (vgvishesh) [inline src/dto/get-api-key.dto.ts:21]: email verify bhi kar lete h. LEts just use a regex (regular expressoin) to check this.

File: src/dto/get-api-key.dto.ts (hunk 5 above)
Comment (vgvishesh) [reply src/dto/get-api-key.dto.ts:37]: Sring size limit kar sakte h yahan (optional)

Comment (rex-god) [reply src/dto/get-api-key.dto.ts:37]: Don't know actually. Maybe 3-4

File: src/dto/get-api-key.dto.ts (hunk 6 above)
Comment (rex-god) [reply src/dto/get-api-key.dto.ts:21]: `IsEmail` decorator does that.

**Comment by rex-god** [comment]
ok

**Review by vgvishesh — APPROVED** [review]

File: src/app.service.ts [hunk 7]
```diff
@@ -20,15 +20,18 @@ export class AppService {
     return 'Welcome to explore Shyft APIs!';
   }
 
-  async getApiKey(getApiKeyDto: GetApiKeyDto): Promise<boolean> {
+  async getApiKey(getApiKeyDto: GetApiKeyDto): Promise<any> {
```
Comment (codinpsycho) [inline src/app.service.ts:23]: Shouldn't we return string?

Comment (rex-god) [reply src/app.service.ts:23]: we're returning the object. e.g., { "api_key": "we53ddse332" }

**Pull request merged**